src/lib/mcleece/sodium_crypto_box.h

```diff
@@ -0,0 +1,147 @@
+// refactored/modified from libsodium crypto_box_seal
```

Comment:
This code shares enough in common with its libsodium roots that it feels very suspect to change the license. So I kept it as was.

I'll likely move this into its own subdirectory and out of `mcleece/`.